File: mytime-api/app/models/time_sheet.py
from sqlalchemy import Column, BigInteger, Text, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from typing import List, Optional
from app.core.database import Base


class Timesheet(Base):
    __tablename__ = "Timesheet"

    Id = Column(BigInteger, primary_key=True, index=True)

    FromDate = Column(DateTime,   nullable=True)
    ToDate = Column(DateTime,   nullable=True)
    Description = Column(Text, nullable=True)

    EmployeeId = Column(BigInteger, nullable=True)
    UserId = Column(BigInteger, nullable=True)
    Status = Column(Text, nullable=True)

    AssignedOn = Column(DateTime,   nullable=True)
    AssignedTo = Column(BigInteger, nullable=True)

    ApprovedOn = Column(DateTime,   nullable=True)
    ApprovedBy = Column(BigInteger, nullable=True)
    ApprovedComments = Column(Text, nullable=True)

    CancelledOn = Column(DateTime,   nullable=True)
    CancelledBy = Column(BigInteger, nullable=True)
    CancelledComments = Column(Text, nullable=True)

    RejectedOn = Column(DateTime,   nullable=True)
    RejectedBy = Column(BigInteger, nullable=True)
    RejectedComments = Column(Text, nullable=True)

    CreatedBy = Column(BigInteger, nullable=True)
    CreatedOn = Column(DateTime,   nullable=True)
    ModifiedBy = Column(BigInteger, nullable=True)
    ModifiedOn = Column(DateTime,   nullable=True)

    IsActive = Column(Boolean, nullable=True)

    # One-to-many relationship to TimesheetTask
    timesheet_tasks = relationship("TimesheetTask", back_populates="timesheet")

    def __repr__(self):
        return f"<Timesheet(Id={self.Id}, Status='{self.Status}', EmployeeId={self.EmployeeId})>"

File: mytime-ui/src/app/common/models/timesheet.ts
import { TimesheetTask } from "./timesheet_task";

export interface Timesheet {
  Id?: number;
  FromDate?: Date | string;
  ToDate?: Date | string;
  Description?: string;
  EmployeeId?: number;
  UserId?: number;
  Status?: string;
  AssignedOn?: Date | string;
  AssignedTo?: number;
  ApprovedOn?: Date | string;
  ApprovedBy?: number;
  ApprovedComments?: string;
  CancelledOn?: Date | string;
  CancelledBy?: number;
  CancelledComments?: string;
  RejectedOn?: Date | string;
  RejectedBy?: number;
  RejectedComments?: string;
  TotalHrs?: number;
  CreatedBy?: number;
  CreatedOn?: Date | string;
  ModifiedBy?: number;
  ModifiedOn?: Date | string;
  IsActive?: boolean;
  Tasks?: TimesheetTask[];
}